### ISSUE TITLE
ARGOCD_APP_NAME incompatible with helm release name when apps in any namespace

### ISSUE THREAD

**Issue by romachalm** [argoproj/argo-cd#15354]
# Summary

I configured argocd with applications in any namespace instead of `argocd`, for example, `myproject`
I have avp plugins configured with 
```
helm template . --name-template "${ARGOCD_ENV_HELM_RELEASE_NAME:-$ARGOCD_APP_NAME}" -n $ARGOCD_APP_NAMESPACE ...
```

But when deploying app in other namespaces, argocd prefixes the `ARGOCD_APP_NAME` with the namespace, followed by underscore, ie `myproject_my-app`, instead of having the app name as defined in the application manifest

The plugin fails with the following error 
```
Error: release name \"myproject_my-app\": invalid release name, must match regex ^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$ and the length must not be longer than 53\n
```
ie it does not accept the underscore

# Motivation

The applications in any namespace are interesting for RBAC in multitenant argocd. But this new app name makes the migration first incompatible, and requires to edit each application to add the `ARGOCD_ENV_HELM_RELEASE_NAME` envar. Moreover, it makes the migration non transparent when I move applications to dedicated namespaces, since some helm charts use the helm release to name the resources, making them pruned and recreated

# Proposal

I suggest to have an envar  such `ARGOCD_RAW_APP_NAME` or something similar to have the app name as defined in the application manifest, ie non namespaced, to be used in the plugins without having the compatibility issue

**Comment by romachalm** [comment]
That's what I fnally did in my plugin to unblock. Thanks

For the length issue, I was thinking the name of the app already had the length constrained by k8s, but indeed I am wrong :  since the limit for the name is 63 chars, and not 53, so that would be rejected by helm release name too (and that would be the case even for apps in `argocd` ns) 

I thought this 53 chars was a old limit in k8s. Seems not to be the case in helm 😭 

I see that someone from argocd communicaty already raised the issue in helm project https://github.com/helm/helm/issues/6006#issuecomment-1590915348

Still, what about having an envar with the application name without the namespace ? As defined in 
```yaml
metadata: 
  name: my-app
```

**Comment by jannfis** [comment]
> since the limit for the name is 63 chars
> ...
> I see that someone from argocd communicaty already raised the issue in helm project https://github.com/helm/helm/issues/6006#issuecomment-1590915348

Yes, and as correctly noted in that issue, the problem gets worse. The `name` of a resource may be 253 characters long, while the values for labels are restricted to 63. So when you use the default label tracking method, application names are (artificially) restricted to 63 characters, however, if you use annotation based tracking method, you can use 253 characters for an application name.

> Still, what about having an envar with the application name without the namespace

Fine to me, it wouldn't fully solve the problem for Helm release names though.

**Comment by lusu007** [comment]
I encountered the same issue today using the ArgoCD Vault plugin.

```
time="2023-12-07T14:58:29Z" level=warning msg="Plugin command returned zero output" command="{[sh -c helm template \"$ARGOCD_APP_NAME\" -n \"$ARGOCD_APP_NAMESPACE\" --include-crds . | argocd-vault-plugin generate -] []}" execID=bf134 stderr="Error: release name \"project-name_app-name\": invalid release name, must match regex ^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$ and the length must not be longer than 53\n"
```

**Comment by paulo-tio** [comment]
I'm facing the same issue using vault replacer as sidecar container plugin

**Comment by Harinath-ponnathota** [comment]
@romachalm I am getting the exact same issue. May I know what you did to fix invalid release name
